### PR TITLE
[TE] yaml - onboarding tasks

### PR DESCRIPTION
**Pull request by jihaozh**
Run yaml onboarding tasks after the alert is set up. The onboarding task replays with last month's data and retunes the pipeline against the replay result.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/onboard/YamlOnboardingTaskRunner.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    config.setLastTimestamp(result.getLastTimestamp());
+    this.detectionDAO.update(config);
+
+    for (MergedAnomalyResultDTO anomaly : result.getAnomalies()) {
+      anomaly.setAnomalyResultSource(AnomalyResultSource.ANOMALY_REPLAY);
+      this.anomalyDAO.save(anomaly);
+    }
+
+    for (MergedAnomalyResultDTO mergedAnomalyResultDTO : result.getAnomalies()) {
+      this.anomalyDAO.save(mergedAnomalyResultDTO);
```
Comment (akshayrai) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/onboard/YamlOnboardingTaskRunner.java]: Can this be merged with the above loop?

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/onboard/YamlOnboardingTaskRunner.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+import org.apache.pinot.thirdeye.detection.DetectionPipelineLoader;
+import org.apache.pinot.thirdeye.detection.DetectionPipelineResult;
+import org.apache.pinot.thirdeye.detection.yaml.YamlDetectionConfigTranslator;
+import org.apache.pinot.thirdeye.detection.yaml.YamlDetectionTranslatorLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.Yaml;
+
+
+public class YamlOnboardingTaskRunner implements TaskRunner {
```
Comment (akshayrai) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/onboard/YamlOnboardingTaskRunner.java]: Might include some class level docs explaining what tasks does this onboarding job perform.

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java [hunk 3]
```diff
@@ -225,6 +267,7 @@ public Response createYamlAlert(@ApiParam(value =  "a json contains both notific
     }
     long alertId = Long.parseLong(((Map<String, String>) response.getEntity()).get("detectionAlertConfigId"));
 
+    createYamlOnboardingTask(detectionConfigId, startTime, endTime);
```
Comment (akshayrai) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java]: Enclose in try-catch. If exception is thrown, revert changes and display message 'failed to setup the onboarding job`

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/onboard/YamlOnboardingTaskRunner.java (hunk 1 above)
Comment (jihaozh) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/onboard/YamlOnboardingTaskRunner.java]: merged

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/onboard/YamlOnboardingTaskRunner.java (hunk 2 above)
Comment (jihaozh) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/onboard/YamlOnboardingTaskRunner.java]: added

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java (hunk 3 above)
Comment (jihaozh) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/yaml/YamlResource.java]: sure. good idea.

**Pull request merged**